modules/seqexec/model/src/main/scala/seqexec/model/enum/SingleActionOp.scala

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package seqexec.model.enum
+
+import cats.Eq
+import cats.implicits._
+import gem.Observation
+// import seqexec.model.ClientId
+
+sealed trait SingleActionOp extends Product with Serializable {
+  val sid: Observation.Id
+  val resource: Resource
+}
+
+object SingleActionOp {
+  final case class Started(sid: Observation.Id, resource: Resource)
+    extends SingleActionOp
+  final case class Completed(sid: Observation.Id, resource: Resource)
+    extends SingleActionOp
+  final case class Error(sid: Observation.Id, resource: Resource)
+    extends SingleActionOp
+
+  implicit val equal: Eq[SingleActionOp] = Eq.instance {
```

Comment:
do we have a test of the `Eq` laws?

Reply:
I'll add that. I figured I was probably missing some law testing.
Do you know where is it? I can't seem to find it offhand.

Reply:
`ModelSpec`